snapcraft/internal/repo/_deb.py

```diff
@@ -413,7 +533,7 @@ def _format_sources_list(sources_list, *,
         'release': release,
         'suffix': suffix,
         'security': security,
-    })
+    }).replace('deb', 'deb [arch={}]'.format(deb_arch) if foreign else 'deb')
```

Comment:
I'm curious why we're replacing here as opposed to making it a templated param? We should probably be catching KeyErrors here, by the way, and turning them into warnings since this can come from custom plugins as well.

Assuming we went this route for a good reason, we need to make sure it's robust to sources like `deb http://debian.mycompany.com/[...]`.

Reply:
I sorta considered this internal API used only but Snapcraft itself. But actually you're right, this would benefit from some error handling. The replace was meant to avoid having `[arch=` in there when it's not needed, but I should use an `$arch` variable for that.

Reply:
Or just use a `$deb` variable for the whole thing, yeah, something like that.